PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-springcloud/src/main/java/com/microsoft/azure/toolkit/intellij/springcloud/deplolyment/SpringCloudDeploymentConfiguration.java

```diff
@@ -69,7 +70,7 @@ public void readExternal(Element element) throws InvalidDataException {
                 .orElse(SpringCloudAppConfig.builder().deployment(SpringCloudDeploymentConfig.builder().build()).build());
         Optional.ofNullable(element.getChild("Artifact"))
                 .map(e -> e.getAttributeValue("identifier"))
-                .map(manager::getAzureArtifactById)
+                .map(id -> ReadAction.compute(() -> manager.getAzureArtifactById(id)))
```

Comment:
this fix can be reverted as I have fixed it at "root": https://github.com/microsoft/azure-tools-for-java/commit/ff32305bc9387ca29c1f805401ff4923d16a6f01

we can use `AzureTaskManager#read`, or `#readAsObservable` which handles errors automatically